distribution/lib/Standard/Base/0.0.0-dev/src/Network/HTTP/Response.enso

```diff
@@ -122,7 +123,7 @@ type Response
        format.
     @format decode_format_selector
     decode : File_Format -> Any -> Any
-    decode self format=Auto_Detect ~if_unsupported=(Error.throw (Unsupported_Content_Type.Error self.content_type)) =
+    decode self format=Auto_Detect ~if_unsupported=Throw_Unsupported_Error =
```

Comment:
> See the comment below so ignore, though I wander if we should remove argument for now and make a boolean flag?

I'd like to avoid changing the API because of internal compiler bugs. From the outside it looks and behaves all the same so no one should notice this workaround unless they are browsing the code.

Reply:
I'm still undecided.

It feels like indeed the boolean flag _could_ be enough for us at the moment. It seems to be enough for S3 and likely it will still be enough for some other simple backends.

I'm still not sure if it's better than the current approach which (at least _in theory..._) gives the (hypothetical) new backends more flexibility in how the details of each operation are carried out.

I have a hard time judging if I don't want to switch because I'm attached to my code and the sunken-cost fallacy OR if it's actually good intuition that this kind of flexibility _can_ be useful in the future.

I have a feeling that the current more 'class-based' approach may be more readable (as we explicitly state how the implementation behaves, instead of setting some flags and 'deriving' an implementation based on the combination of these flags' values). But I know I may not be objective here.

Reply:
Given my indecisiveness here, I'd lean on keeping the code as-is for now - given that I think both approaches are essentially OKayish and that I prefer the current one **and** that keeping it is **less work** than rewriting - I think I'd just keep it for now.

But I'm happy to try a rewrite if requested.